lc56.py

```diff
@@ -1 +1,24 @@
+class Solution:
+    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
+        def isOverlap(a,b):
+            maxStart = max(a[0],b[0])
+            minFinish = min(a[1],b[1])
+            return (minFinish >=  maxStart) # overlap
+        
+        intervals.sort(key = lambda x:x[0])
+        res = []
+        temp = intervals[0]
+        res.append(temp)
+        
+        for i in range(1,len(intervals)):
+            if isOverlap(temp,intervals[i]):
+                res[i-1] = None
```

Comment:
Looks like you're setting the previous result val to None that you just computed (temp). Why not just update temp[1] to be max(temp[1], maxFinish). temp[0] will always be minStart. In actually I don't think you need a separate variable for temp, as temp will always be the last element of res so just use res[-1].